xds/internal/client/client_callback.go

```diff
@@ -88,11 +88,20 @@ func (c *Client) newLDSUpdate(d map[string]ldsUpdate) {
 			c.ldsCache[name] = update
 		}
 	}
-	// TODO: handle removing resources, which means if a resource exists in the
-	// previous update, but not in the new update. This needs the balancers and
-	// resolvers to handle errors correctly.
-
-	// TODO: remove item from cache and remove corresponding RDS cached data.
+	for name := range c.ldsCache {
+		if _, ok := d[name]; !ok {
```

Comment:
Can we have a better variable name for `d` whose scope has now gotten bigger.

Reply:
Renamed to `updates`